### PR TITLE
Adds Truncate to platform.File

### PR DESCRIPTION
**Pull request by codefromthecrypt**
See #1417

### CODE REVIEW

File: internal/sys/fs.go [hunk 1]
```diff
@@ -273,7 +283,6 @@ type FileEntry struct {
 	// was called.
 	ReadDir *ReadDir
 
-	openPath string
```
Comment (codefromthecrypt) [inline internal/sys/fs.go:276]: this is no longer needed to be stored externally

**Comment by codefromthecrypt** [comment]
looks like there's disagreement in the syscall.Errno returned when ftruncate acts on a directory not a file. I'll sort it out and mark draft meanwhile

**Comment by codefromthecrypt** [comment]
meh fix one another breaks (windows). I'll mark draft again

**Comment by evacchi** [comment]
apparently the code on `main` can't acquire this lock, which trickles down into the original error. Your recent change is now acquiring the lock, so it returns with "use of closed file"; code on this branch succeeds in acquiring the lock

```
// fd_windows.go:
func (fd *FD) Write(buf []byte) (int, error) {
	if err := fd.writeLock(); err != nil {
		return 0, err
	}

// the callee returns the actual error
func (fd *FD) writeLock() error {
	if !fd.fdmu.rwlock(false) {
		return errClosing(fd.isFile)
	}
	return nil
}

```

now the question is whether the behavior on main is actually correct

**Comment by codefromthecrypt** [comment]
will rework this after #1430

**Review by evacchi — APPROVED** [review]

**Pull request merged**